### PR TITLE
Rework View Filters and Sortings

### PR DESCRIPTION
**Pull request by jpaddison3**
Fix bugs in the process.

First part of a series of steps on the way to the Sort By Timeframe feature. This is WIP because before I go writing a migration to change the name of a user setting, I'd like to confirm that you think this is reasonable.

Previously in PostsListSettings we were using views as the method of sorting. Views are a powerful tool however, and they frequently want to overwrite the filter. This PR introduces the concept of sortings, which can be nicely composed with filters.

There are two TODO[WIP] comments that I will complete up your go-ahead. There is one TODO-Q comment, where I'm curious whether you think that allPosts should have stickied content in some of the sortings. It looks like there's a merge conflict that's crept in. I'll deal with that, again on your go ahead.

### CODE REVIEW

**Comment by Raemon** [comment]
The word "view" isn't arbitrary – it's mapping to the views.js file. My sense is that probably "views" should stay the same, and the new category that you should add is something like "groupBy" or something, where the three choices are:

– Monthly
– Daily
– List

So most of the "view" functionality would just stay the same (except for removing 'daily' from it), and instead have an additional set of functionality under GroupBy

**Comment by jpaddison3** [comment]
This PR makes no attempt at attacking the Daily / Monthly etc split. It softens the ground by noticing that `view`s as used by Vulcan are not very composable. We can still use views where there's a clear single use-case like rss feeds, but for flexible things like allPosts, we should use `filter` and `sortedBy`.

File: packages/lesswrong/lib/collections/posts/views.js [hunk 1]
```diff
@@ -156,7 +221,7 @@ ensureIndex(Posts,
 
 
 Posts.addView("top", terms => ({
-  options: {sort: setStickies({baseScore: -1}, terms)}
+  options: {sort: setStickies(sortings["tops"], terms)}
```
Comment (jimrandomh) [inline packages/lesswrong/lib/collections/posts/views.js]:
```suggestion
  options: {sort: setStickies(sortings.tops, terms)}
```

**Comment by jimrandomh** [comment]
- [ ] Nitpick with `sortings["foo"]` vs `sortings.foo`
- [ ] The migration that you mentioned for user-profile saved sort criteria
- [ ] Take out the debug logs

**Comment by Raemon** [comment]
Also, I think by "tops" you mean "top"

**Comment by jpaddison3** [comment]
I need to
  a) Fix merge conflicts
  b) Retest all the things

Otherwise I claim this is now ready to go, subject to review of the migration.

**Comment by jpaddison3** [comment]
Note that it shouldn't be merged until the current devel->master PR and deploy is done. This should probably spend some time on lessestwrong.

**Comment by jpaddison3** [comment]
Alright, merge complete and testing looks good.

**Pull request merged**